### PR TITLE
Automatically fix incorrect checkboxes

### PR DESCRIPTION
**Pull request by revolter**
Related to https://github.com/fastlane/fastlane/pull/11214.

### CODE REVIEW

**Comment by revolter** [comment]
How could I skip updating if the text didn't change?

File: bot.rb [hunk 1]
```diff
@@ -109,9 +109,13 @@ def process_open_issue(issue)
       bot_actions = []
       process_inactive(issue)
 
-      return if issue.comments > 0 # there maybe already some bot replys
+      # return if issue.comments > 0 # there maybe already some bot replies
       bot_actions << process_code_signing(issue)
-      bot_actions << process_env_check(issue)
+      # bot_actions << process_env_check(issue)
```
Comment (KrauseFx) [inline bot.rb]: Let's remove these lines if we don't need them any more

**Comment by KrauseFx** [comment]
We usually use tags or check if there is already at least one comment from the `issue-bot` on a particular issue.

File: bot.rb (hunk 1 above)
Comment (revolter) [reply bot.rb]: Sorry, forgot to uncomment it. I commented out for easier testing.

**Comment by revolter** [comment]
This doesn't work, as this just edits the original post. I was looking for a terser way to do:

```
new_body = fix_checkboxes(issue.body)

if (new_body != issue.body) {
    client.update_issue(SLUG, issue.number, issue.title, new_body)
}
```

**Comment by milch** [comment]
Is there a reason you're not doing that (checking `new_body != issue.body`)?

**Comment by revolter** [comment]
@milch, Do you have some time to review my last comment? I tested it and it seemed to work properly (https://github.com/revolter/fastlane-bot-test/issues/2).

File: bot.rb [hunk 2]
```diff
@@ -394,5 +402,11 @@ def collect_pr_references_from_line(line, release_name, prs_to_releases)
         end
       end
     end
+
+    def fix_checkboxes(text)
+      new_text = text.gsub(/^- \[\s*\S+\s*\]/, "- [x]")
+
+      return new_text if new_text != text else nil
```
Comment (milch) [inline bot.rb]: What the hell, that works?!

**Comment by milch** [comment]
Ah, I'm a pleb in this repo. My review counts for nothing. Sorry ☹️

File: bot.rb (hunk 2 above)
Comment (revolter) [reply bot.rb]: Why not?

**Comment by revolter** [comment]
No problem, didn't want specifically a GitHub review, but at least another pair of eyes 😄

File: bot.rb (hunk 2 above)
Comment (revolter) [reply bot.rb]: It's a ternary operator [using keywords instead of operators](https://stackoverflow.com/a/4252945/865175).

Comment (milch) [reply bot.rb]: Interesting, makes sense. Although the order is off compared to the ternary operator:

```ruby
return (new_text != text ? new_text : nil) 
```

Reads kind of like a sentence this way, though. I like it.

Comment (milch) [reply bot.rb]: Also from that link I would've thought the order would need to be 

```ruby
return if new_text != text then new_text else nil
```

Comment (revolter) [reply bot.rb]: Oh, right, I got it mixed up with [Python's way](https://stackoverflow.com/a/2802748/865175). I don't know how it worked though :|

Comment (revolter) [reply bot.rb]: It looks like `else nil` was somehow ignored, because [this](https://stackoverflow.com/a/42308805/865175) is the correct syntax.

**Pull request merged**